// react-layout-components/lib/layouts/Border.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; desc = parent = getter = undefined; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _assignStyles = require('assign-styles');

var _assignStyles2 = _interopRequireDefault(_assignStyles);

var _componentsBox = require('../components/Box');

var _componentsBox2 = _interopRequireDefault(_componentsBox);

var BorderLayout = (function (_React$Component) {
  _inherits(BorderLayout, _React$Component);

  function BorderLayout() {
    _classCallCheck(this, BorderLayout);

    _get(Object.getPrototypeOf(BorderLayout.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(BorderLayout, [{
    key: 'render',
    value: function render() {
      var map = {
        north: [],
        west: [],
        south: [],
        east: []
      };
      var center = this.props.children.map(function (box) {
        var newProps = box.props;
        newProps.flex = 1;
        if (box.props.region && map[box.props.region]) {
          map[box.props.region].push(_react2['default'].cloneElement(box, newProps, box.props.children));
        } else {
          newProps.flex = 1;
          return _react2['default'].cloneElement(box, newProps, box.props.children);
        }
      });

      return _react2['default'].createElement(
        _componentsBox2['default'],
        { column: true, fit: true },
        _react2['default'].createElement(
          _componentsBox2['default'],
          { row: true },
          map.north
        ),
        _react2['default'].createElement(
          _componentsBox2['default'],
          { row: true, flex: '1' },
          _react2['default'].createElement(
            _componentsBox2['default'],
            null,
            map.west
          ),
          _react2['default'].createElement(
            _componentsBox2['default'],
            { flex: '1' },
            center
          ),
          _react2['default'].createElement(
            _componentsBox2['default'],
            null,
            map.east
          )
        ),
        _react2['default'].createElement(
          _componentsBox2['default'],
          { row: true },
          map.south
        )
      );
    }
  }]);

  return BorderLayout;
})(_react2['default'].Component);

exports['default'] = BorderLayout;
module.exports = exports['default'];
